internal/services/oidc.go

```diff
@@ -106,33 +115,75 @@ func (osvc *OIDCService) IsFlowSecure(state string, token models.Tokens) (bool,
 	return true, nil
 }
 
-// CreateJWT creates a new token and the claims you would like it to contain.
+// CreateJWT creates a new JWT token and Custom Claims.
 func (osvc *OIDCService) CreateJWT(cc *models.CustomClaims) (models.CustomClaims, error) {
-	// For HMAC signing method, the key can be any []byte
-	hmacRandSecret, err := randomByte(1990)
-	if err != nil {
-		osvc.logger.Errorf("An error occurred while generating HMAC. Error: %v", err)
-		return models.CustomClaims{}, err
-	}
+	var err error
 
 	customClaims := models.CustomClaims{
 		Name:    cc.Name,
 		Email:   cc.Email,
 		Profile: cc.Profile,
 		RegisteredClaims: jwt.RegisteredClaims{
-			ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Duration(osvc.client.Configuration.JWTExpires))),
+			ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Duration(osvc.configuration.UserJWTExpires))),
 			IssuedAt:  jwt.NewNumericDate(time.Now()),
 		},
 	}
 
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, customClaims)
 
 	// Sign and get the complete encoded token as a string using the secret.
-	customClaims.AccessToken, err = token.SignedString(hmacRandSecret)
+	customClaims.AccessToken, err = token.SignedString(hmacSecret)
 
 	return customClaims, err
 }
 
+// ValidateJWT validates the token and returns the claims.
+// even if there is an error, return the invalid token instead of nil.
+func (osvc *OIDCService) ValidateJWT(tokenString string) (map[string]interface{}, error) {
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		// Don't forget to validate the alg is what you expect:
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+
+		return hmacSecret, nil
+	})
+
+	if token == nil {
+		return invalidToken, errors.New("an unexpected error occurred while validating the JWT token")
+	}
+
+	if token.Valid {
+		claims := token.Claims.(jwt.MapClaims)
+		claims["access_token"] = tokenString
+
+		return claims, nil
```

Comment:
this is not a validation

Reply:
i didn't catch that...

Reply:
we changed the name from `ValidateJWT` to `Decode`.